posts/posts-router.js

```diff
@@ -0,0 +1,118 @@
+const express = require('express');
+
+const Hubs = require('./db.js');
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+    try {
+        const hubs = await Hubs.find();
+        res.status(200).json(hubs);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The posts information could not be retrieved.',
+        });
+    }
+});
+
+
+router.get('/:id', async (req, res) => {
+    try {
+      const hub = await Hubs.findById(req.params.id);
+  
+      if (hub) {
+        res.status(200).json(hub);
+      } else {
+        res.status(404).json({ message: 'Hub not found' });
+      }
+    } catch (error) {
+      // log error to database
+      console.log(error);
+      res.status(500).json({
+        message: 'The post with the specified ID does not exist.',
+      });
+    }
+  });
+  
+  router.get('/:id/comments', (req, res) => {
+    const { id } = req.params;
+  
+    Hubs.findHubMessages(id)
+      .then(messages => {
+        if (messages.length > 0) {
+          res.status(200).json(messages);
+        } else {
+          res.status(404).json({ message: 'The post with the specified ID does not exist.' });
+        }
+      })
+      .catch(err => {
+        res.status(500).json({
+          message: 'Error retrieving the hub',
+        });
+      })
+  })
+  
+  router.post('/:id/comments', async (req, res) => {
+      const messageInfo = { ...req.body, hub_id: req.params.id };
+      try {
+        const message = await Hubs.addMessage(messageInfo);
+        res.status(201).json(message);
+      } catch (error) {
+        console.log(error);
```

Comment:
The .catch should be used for generic errors or unexpected system errors, and not for not found resources. I'd suggest to run a check to find the resource matching the id in the try block and return the status code 404 there, and use the catch block for status 500 and real error handling. Non-existent resources are not actually errors.